src/errors/applicationErrors.js

```diff
@@ -96,6 +96,16 @@ class SilentError extends BusinessError {
 	}
 }
 
+class SyncError extends Error {
+	constructor(taskType, syncId, error) {
+		super(taskType);
+		this.name = this.constructor.name;
+		this.message = error.message;
+		this.error = error;
```

Comment:
We should look if we really need the complete error inside of this, or only keys that we put to top level like message.